src/pkgcheck/checks/rust.py

```diff
@@ -45,3 +60,21 @@ def feed(self, pkg: bash.ParseTree):
                             pkg=pkg,
                         )
                         break
+        for node, _ in bash.cmd_query.captures(pkg.tree.root_node):
+            call_name = pkg.node_str(node.child_by_field_name("name"))
+            if call_name == "cargo_crate_uris":
+                row, _ = node.start_point
+                line = pkg.node_str(node.parent)
+                if node.child_count == 1 or (
+                    node.child_count == 2
+                    and any(
+                        pkg.node_str(var_node) == "CRATES"
+                        for var_node, _ in bash.var_query.captures(node.children[1])
+                    )
```

Comment:
I'm not sure if I understand this correctly but the replacement works only if `${CRATES}` (or `"${CRATES}"`) is the only argument.

Reply:
Yes, that is why I verify the node count is exactly 2. I've even added a test for `${CRATES} something` and verify it ignores it :)

Reply:
Well, I guess I don't understand what's the iteration for but I guess it's high magic.